File: src/routers/projects.py
from fastapi import APIRouter, HTTPException
from pydantic import BaseModel
from config import supabase

router = APIRouter(prefix="/projects", tags=["projects"])


class ProjectRequest(BaseModel):
    user_id: str
    nombre: str
    descripcion: str


class RenameRequest(BaseModel):
    nombre: str


@router.get("/{user_id}")
def get_projects(user_id: str):
    projects = supabase.table("projects").select("*").eq("user_id", user_id).execute()
    return {"projects": projects.data}


@router.post("/")
def create_project(request: ProjectRequest):
    project = (
        supabase.table("projects")
        .insert(
            {
                "user_id": request.user_id,
                "nombre": request.nombre,
                "descripcion": request.descripcion,
            }
        )
        .execute()
    )
    return {"project_id": project.data[0]["id"]}


@router.patch("/{project_id}")
def rename_project(project_id: str, request: RenameRequest):
    supabase.table("projects").update({"nombre": request.nombre}).eq("id", project_id).execute()
    return {"status": "updated"}


@router.delete("/{project_id}")
def delete_project(project_id: str):
    supabase.table("projects").delete().eq("id", project_id).execute()
    return {"status": "deleted"}

File: mobile/src/services/projects.ts
import { API_URL } from './config'

export interface Project {
  id: string
  nombre: string
  descripcion: string
  user_id: string
  created_at?: string
}

export async function getProjects(userId: string): Promise<Project[]> {
  const res = await fetch(`${API_URL}/projects/${userId}`)
  if (!res.ok) throw new Error('Error al cargar proyectos')
  const data = await res.json()
  return data.projects
}

export async function createProject(
  userId: string,
  nombre: string,
  descripcion: string
): Promise<string> {
  const res = await fetch(`${API_URL}/projects/`, {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify({ user_id: userId, nombre, descripcion }),
  })
  if (!res.ok) throw new Error('Error al crear proyecto')
  const data = await res.json()
  return data.project_id
}

export async function renameProject(projectId: string, nombre: string): Promise<void> {
  const res = await fetch(`${API_URL}/projects/${projectId}`, {
    method: 'PATCH',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify({ nombre }),
  })
  if (!res.ok) throw new Error('Error al renombrar proyecto')
}

export async function deleteProject(projectId: string): Promise<void> {
  const res = await fetch(`${API_URL}/projects/${projectId}`, { method: 'DELETE' })
  if (!res.ok) throw new Error('Error al eliminar proyecto')
}
